app/bundles/LeadBundle/Entity/FrequencyRuleRepository.php

```diff
@@ -19,69 +19,46 @@
-    {
-        $q = $this->_em->getConnection()->createQueryBuilder();
-
-        $selectFrequency = ($defaultFrequencyNumber) ? 'IFNULL(fr.frequency_number,:defaultNumber) as frequency_number' : 'fr.frequency_number';
-        $selectNumber    = ($defaultFrequencyTime) ? 'IFNULL(fr.frequency_time,:frequencyTime) as frequency_time' : 'fr.frequency_time';
-
-        $q->select("ch.$statContactColumn, $selectFrequency, $selectNumber")
-            ->from(MAUTIC_TABLE_PREFIX.$statTable, 'ch')
-            ->leftJoin('ch', MAUTIC_TABLE_PREFIX.'lead_frequencyrules', 'fr', "ch.{$statContactColumn} = fr.lead_id");
-
-        if ($channel) {
-            $q->andWhere('fr.channel = :channel or fr.channel is null')
-                ->setParameter('channel', $channel);
-        }
-
-        if (!empty($defaultFrequencyTime)) {
-            $q->andWhere('ch.'.$statSentColumn.' >= case fr.frequency_time 
-                    when \'MONTH\' then DATE_SUB(NOW(),INTERVAL 1 MONTH) 
-                    when \'DAY\' then DATE_SUB(NOW(),INTERVAL 1 DAY) 
-                    when \'WEEK\' then DATE_SUB(NOW(),INTERVAL 1 WEEK)
-                    else DATE_SUB(NOW(),INTERVAL 1 '.$defaultFrequencyTime.')
-                    end')
-                ->setParameter('frequencyTime', $defaultFrequencyTime);
-        } else {
-            $q->andWhere('(ch.'.$statSentColumn.' >= case fr.frequency_time
-                     when \'MONTH\' then DATE_SUB(NOW(),INTERVAL 1 MONTH)
-                     when \'DAY\' then DATE_SUB(NOW(),INTERVAL 1 DAY)
-                     when \'WEEK\' then DATE_SUB(NOW(),INTERVAL 1 WEEK)
-                    end)');
-        }
-
+    public function getAppliedFrequencyRules(
+        $channel,
+        $leadIds,
+        $defaultFrequencyNumber,
+        $defaultFrequencyTime,
+        $statTable = 'email_stats',
+        $statContactColumn = 'lead_id',
+        $statSentColumn = 'date_sent'
+    ) {
         if (empty($leadIds)) {
-            // Preventative for fetching every single email stat
-            $leadIds = [0];
+            return [];
         }
-        $q->andWhere(
-            $q->expr()->in("ch.$statContactColumn", $leadIds)
-        );
 
-        $q->groupBy("ch.$statContactColumn, fr.frequency_time, fr.frequency_number");
-
-        $havingAnd = 'AND '.$q->expr()->in("ch.$statContactColumn", $leadIds);
-        if ($defaultFrequencyNumber != null) {
-            $q->having("(count(ch.$statContactColumn) >= IFNULL(fr.frequency_number,:defaultNumber) $havingAnd)")
-                ->setParameter('defaultNumber', $defaultFrequencyNumber);
-        } else {
-            $q->having("(count(ch.$statContactColumn) >= fr.frequency_number $havingAnd)");
+        $violations = $this->getCustomFrequencyRuleViolations($channel, $leadIds, $statTable, $statContactColumn, $statSentColumn);
+
+        if ($defaultFrequencyTime && $defaultFrequencyTime) {
```

Comment:
Is this condition supposed to be `&&` of `$defaultFrequencyTime` and itself, or should it be `($defaultFrequencyNumber && $defaultFrequencyTime)` ?

Reply:
Supposed to be both.